### DISCUSSION TITLE
"Best" battery solution in 2024 for Pi 3

### DISCUSSION THREAD

**Discussion by jnettels** [MiczFlor/RPi-Jukebox-RFID#2270] [Q&A] [answered]
Hi there,

I already found many discussions around this topic. However, despite many hours of search, I was not satisfied with the results.
I am compiling a list of options I identified and why they may not work for me. I hope this helps my sanity and maybe someone else, too. If you have a solution for me, I would be very thankful!

I am running a Raspberry Pi 3 Model B. I was thinking about upgrading to Pi 5 but I guess the new power requirements make it even more complicated. I am located in Germany.

The goal: Allow the phoniebox to be powered by a battery which can be loaded while the system is running and use an external power button to power on / shut down the system.

There are two main categories:
- Combining a On/Off SHIM with a consumer powerbank
- Installing a dedicated Raspberry Pi UPS

### On/Off SHIM plus consumer powerbank
This seems to be the most recommended option here in the finished projects.

The recommended on/off SHIM is: https://www.amazon.de/gp/product/B072FDDKSD

However, finding a suitable powerbank seems tricky. It needs to support "pass through" of power from input to output, which means that the pi can always stay connected. But even for those product that may support it, you often find comments saying in the moment of connection, the pi reboots because of a voltage drop. There are also comments that say that some powerbanks go into standby after the pi has been shut down, which prevents it from being turned back on again.

There is one that I found recommended here, but it seems to be no longer available:
- https://www.amazon.de/gp/product/B01422TC14

👍 Recommendend and available:
- https://www.amazon.de/gp/product/B0B17SZRG8
  - Has USB-C power input. ( For me, this was not really clear from the product pictures)
  - I have now used it myself for a couple of days and it seems to be working as expected and fits all my demands.
  - Have not stress tested the battery life, but 10.000 mAh should last a while!

Here is a long list of "pass through" power banks:
https://www.usb-akku.guru/-Rangliste/Pass-Through-Charging
For many of them, I cannot find a place to buy.

Those powerbanks typically have a large form factor. And there is no possibility to integrate the charing status into the phoniebox system UI.

### Dedicated Raspberry Pi UPS
This should theoretically allow for more "elegant" solutions, where battery percentage readings could be made available in the phoniebox UI. Also logic board and actual battery can be seperate items, which can result in smaller form factor / less space required.

## Pisugar 3 Plus
- https://www.amazon.de/dp/B09QRS666Y
- Respository with documentation available: https://github.com/PiSugar/PiSugar/wiki/PiSugar-3-Series
- Low 5000 mAh capacity, especially considering cost of 50€
- External button can be installed. But to power on, a "short press followed by a long press" is required. It seems like this setting can be changed, but that setting resets everytime the device is turned completely off. I do not think this is usable for a child.
- Several reports of various other issues in the repository

## Seamuing Raspberry Pi 4 UPS Hut 
- Link: https://www.amazon.de/-/en/Seamuing-Raspberry-Pi-Extension-Battery-red/dp/B09LCBQVF5
- Documentation: https://github.com/rcdrones/UPSPACK_V3/blob/master/README_en.md
- 10.000 mAh capacity (battery can be disconnected and replaced)
- Allows an external button (must be a "latching" switch which means first press is "on", next press is "off". I would prefer a "momentary" switch where short press is "on", long press is "off"). Would require e.g. these parts:
  - https://www.amazon.de/gp/product/B07G17LT8Y
  - https://www.amazon.de/gp/product/B01DU9P03Y/ref=ox_sc_act_title_1?smid=AMUN6OW4OKOC5&psc=1
- Awesome offer for 36€. But it is simply not available on Amazon Germany

## DFRobot Raspberry Pi UPS HAT
- Link: https://eu.robotshop.com/de/products/raspberry-pi-ups-hat
- Documentation: https://wiki.dfrobot.com/Raspberry_Pi_UPS_HAT_SKU_DFR0494
- I do not know if an external power button can be connected
- Uses a Micro-USB plug for power input. I want to use a USB-C connector as power input

## Waveshare UPS HAT
- Links:
  - https://www.amazon.de/Waveshare-UPS-HAT-Unterbrechungsfreie-Ausgangsleistung-EN/dp/B08BRPLY15
  - https://www.amazon.de/UPS-HAT-Raspberry-Uninterruptible-Communication/dp/B08C53K14G
- Seems nice, but uses a barrel plug for power input. I want to use a USB-C connector as power input

## Geekworm Raspberry Pi UPS, X728
- 50€ base board: https://www.amazon.de/gp/product/B087J7WTYM
- Very detailed documentation: https://wiki.geekworm.com/X728
- Two li-ion batteries of type 18650 need to be bought separetely.
  - +30€ with this result: https://www.amazon.de/2xSamSung-35E-3450mAh-Flachpol-Akku-Kunststoffbox/dp/B0CT8HGLCD
  - Only 2x 3400 mAh capacity. 
  - But they can easily be swapped if they degrade over time
  - Documentation says those must not have "built-in protection circuit", but I have no idea how to know which batteries this applies to.
  - To be fair: There is a recommended battery type: https://wiki.geekworm.com/File:NCR18650B.pdf
- Very small form factor
- Positive Amazon reviews
- External power button (momentary switch) can be connected
- According to the documenation wiki, a "X728-A2" 2-cell battery holder exists, which can be easily connected to expand capacity. However I can not find a place to order it from
  - I am wondering if other battery packs can be connected

I think the Geekworm Raspberry Pi UPS, X728 is my best option, since it fulfills all my criteria. But 80€ seems quite expensive for about 7000mAh capacity...
I would love to hear your advice!

**Comment by s-martin** [comment]
Thanks for the comprehensive overview.

I think this may help other users.

**Comment by pabera** [comment, marked as answer]
Just to add to the consumer side of things. I have used this Battery Model in both of the Phonieboxes I have built. It allows you to be charged while playing as well.

* https://www.amazon.de/gp/product/B0B17SZRG8

**Reply by jnettels** [reply]
Thanks for the comment, I should have asked here sooner :-) I ordered the battery and will give it a try!

**Reply by mydnic** [reply]
@pabera your link is broken now, would you mind sharing the product reference ? thanks !

**Reply by stopfkuchen** [reply]
I recommend the MupiHAt
you can buy it here: mupihat.de
[mupihat.de]()

**Reply by pabera** [reply]
@mydnic Most of the powerbanks available today should work. The product I used doesn't ship anymore. I'd recommend checking out the form factor (length, height) as those things were important to me to fit it in my device.

**Comment by stopfkuchen** [comment]
Hi, 
there is another solution, which is now available

[MupiHAT.de](https://mupihat.de/)

**Features at a Glance**
- Wide Power Input Range 3.6V to 12V via USB-C or PinHeader 
- Supplies the Raspberry Pi (up to 7 W per default supported without additional heat sink) 
- Fully integrated charger for Lithium Batteries with Protection Features
- UPS Function - seamless plug/un-plug of power
- On/Off Shim Function Push-Button Power-On and clean RPI shutdown
- 2 x 3 W Class-D Audio Amplifier  4-8 Ohm Speakers
- 8 GPIO via PinHeader for connecting auxillaries or LEDs 

 more details on [MupiHAT.de](https://mupihat.de/)

**Comment by pabera** [comment]
@stopfkuchen Thanks for sharing your solution here! It would be helpful if you could present your personal advertisement in English, the common language of our community.

**Reply by stopfkuchen** [reply]
sure. done.

**Comment by X-dark** [comment]
Model advised by @pabera is no longer available. Any alternative?

The Baseus one recommanded [here](https://splittscheid.de/akku-powerbank-einschalter-raspberry-pi/) is available but bulkier/heavier as it is 10000mAH

**Comment by pabera** [comment]
I think any of the available USB-C Power Banks will work. It depends very much on the form factor. I have chose this model mainly because it had to fit my Phoniebox-Wrapper, which it nicely did.

**Reply by X-dark** [reply]
Thanks but what about issues with non-passthrough, battery going in sleep mode once the rpi is powered down,... You think this is less of an issue nowadays?

**Comment by antondd** [comment]
Nice overview! Since I am in Canada, I tend to the one from HighPi/Pishop (HighPi is a brand of this Canadian shop): https://www.pishop.ca/product/raspberry-pi-ups-hat/
Gives enough power for a safe shutdown, and reboots when power comes back. I don't need a long runtime on battery, just want to avoid SD card corruption (or database damage) if power cuts, so a few minutes is enough.

**Comment by umfk** [comment]
What exactly is the problem with directly connecting a powerbank to the Pi without an OnOff SHIM? Is it only the fact that the Pi stays in standby mode and therefore slowly drains the powerbank? If yes, what kind of drain are we talking about? Would a 10 Ah be empty in a matter of hours/days/weeks? And am I correct in assuming that the normal GPIO on/off button should continue to function this way since the Pi stays in standby mode?

Our Phoniebox will be plugged in most of the time like it is currently, I just want to add the possibility for the kids to carry it to a different room/corner and listen to something for some time before plugging it back in.
